src/serializers/encoder/v7/encode_graph.c

```diff
@@ -43,28 +42,96 @@ static void _RdbSaveHeader(RedisModuleIO *rdb, GraphContext *gc) {
 	// Number of keys
 	RedisModule_SaveUnsigned(rdb, GraphEncodeContext_GetKeyCount(gc->encoding_context));
 
-	// Payload type
-	RedisModule_SaveUnsigned(rdb, GraphEncodeContext_GetEncodeState(gc->encoding_context));
 }
 
-// Select the first state to encode.
-static void _SelectFirstState(GraphContext *gc) {
-	// If there are nodes
-	if(Graph_NodeCount(gc->g) > 0) {
-		GraphEncodeContext_SetEncodeState(gc->encoding_context, NODES);
-	} else if(array_len(gc->g->nodes->deletedIdx) > 0) {
-		// If all nodes are deleted
-		GraphEncodeContext_SetEncodeState(gc->encoding_context, DELETED_NODES);
+// Returns the a state information regarding the number of entities required to encode in this state.
+static PayloadInfo _StatePayloadInfo(GraphContext *gc, EncodeState state,
+									 uint64_t offset, uint64_t entities_to_encode) {
+	uint64_t required_entities_count;
+	switch(state) {
+	case NODES:
+		required_entities_count = Graph_NodeCount(gc->g);
+		break;
+	case DELETED_NODES:
+		required_entities_count = Graph_DeletedNodeCount(gc->g);
+		break;
+	case EDGES:
+		required_entities_count = Graph_EdgeCount(gc->g);
+		break;
+	case DELETED_EDGES:
+		required_entities_count = Graph_DeletedEdgeCount(gc->g);
+		break;
+	case GRAPH_SCHEMA:
+		required_entities_count = 0;
+		break;
+	default:
+		assert(false && "Unkown encoding state in _CurrentStatePayloadInfo");
+	}
+	PayloadInfo payload_info;
+	payload_info.state = state;
+	/* When this state will be encoded, the number of entities to encode is the minimum between the number of entities to encode and
+	 * the remaining entities left to encode from the same type. */
+	payload_info.entities_count = MIN(entities_to_encode, required_entities_count - offset);
+	return payload_info;
+}
+
+// This function save the key content schema and returns it so the encoder can now how to encode the key.
+static PayloadInfo *_RdbSaveKeySchema(RedisModuleIO *rdb, GraphContext *gc) {
+	/* Format:
+	*  #Number of payloads info - N
+	*  N * Payload info:
+	*      Encode state
+	*      Number of entities encoded in this state.
+	*/
+	PayloadInfo *payloads = array_new(PayloadInfo, 1);
+	// Get current encoding state.
+	EncodeState current_state = GraphEncodeContext_GetEncodeState(gc->encoding_context);
+	// If it is the start of the encodeing, set the state to be NODES.
+	if(current_state == INIT) current_state = NODES;
+	uint64_t remaining_entities = Config_GetModuleConfig().entities_threshold;
```

Comment:
Introduce `Config_GetNumeric(configuration)` and `Config_GetString(configuration)`